src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -31,63 +36,88 @@ public ApiCompatRunner(string noWarn, (string, string)[] ignoredDifferences, boo
+                    (MetadataInformation rightAssembly, string header) rightTuple = _dict[left][i];
 
-                    IEnumerable<CompatDifference> differences = _differ.GetDifferences(leftSymbols, rightSymbols, leftName: leftName, rightName: apicompatTuples.rightAssemblyRelativePath);
+                    _log.LogMessage(MessageImportance.Low, rightTuple.header);
 
-                    foreach (CompatDifference difference in differences)
+                    foreach (CompatDifference difference in diff.differences)
                     {
                         _log.LogError(
                             new Suppression
                             {
                                 DiagnosticId = difference.DiagnosticId,
                                 Target = difference.ReferenceId,
-                                Left = apicompatTuples.leftAssemblyRelativePath,
-                                Right = apicompatTuples.rightAssemblyRelativePath,
-                                IsBaselineSuppression = isBaselineSuppression
+                                Left = left.AssemblyId,
+                                Right = rightTuple.rightAssembly.AssemblyId,
+                                IsBaselineSuppression = _isBaselineSuppression
                             },
                             difference.DiagnosticId,
                             difference.Message);
                     }
                 }
             }
-            _queue.Clear();
+            _dict.Clear();
         }
 
         /// <summary>
         /// Queues the api compat for 2 assemblies.
         /// </summary>
-        /// <param name="leftPackagePath">Path to package containing left assembly.</param>
-        /// <param name="leftRelativePath">Relative left assembly path in package.</param>
-        /// <param name="rightPackagePath">Path to package containing right assembly.</param>
-        /// <param name="rightRelativePath">Relative right assembly path in package.</param>
-        /// <param name="assemblyName">The name of the assembly.</param>
-        /// <param name="compatibilityReason">The reason for assembly compatibilty.</param>
+        /// <param name="leftMetadataInfo">Metadata information for left assembly.</param>
+        /// <param name="rightMetdataInfo">Metadata information for right assembly.</param>
         /// <param name="header">The header for the api compat diagnostics.</param>
-        public void QueueApiCompat(string leftPackagePath, string leftRelativePath, string rightPackagePath, string rightRelativePath, string assemblyName, string compatibilityReason, string header)
+        public void QueueApiCompat(MetadataInformation leftMetadataInfo, MetadataInformation rightMetdataInfo, string header)
+        {
+            if (_dict.ContainsKey(leftMetadataInfo))
+            {
+                _dict[leftMetadataInfo].Add((rightMetdataInfo, header));
```

Comment:
Maybe the header can just be a resource formatted string and we can create it with the info we have when logging it out? This way we don't have to pass it around?